### ISSUE TITLE
[rmodels] Loading GLB/GLTF fails to parse or render when running on a Big Endian system

### ISSUE THREAD

**Issue by nbe1233** [raysan5/raylib#4647]
Please, before submitting a new issue verify and check:

 - [ x] I tested it on latest raylib version from master branch
 - [ x] I checked there is no similar issue already reported
 - [ x] I checked the documentation on the [wiki](https://github.com/raysan5/raylib/wiki)
 - [ x] My code has no errors or misuse of raylib

### Issue description
glb files fails to load as cgltf.h uses endian-specific code.
gltf files are successfully parsed, but mesh vertices, normals, and indices are incorrectly loaded by  rmodels.c

### Environment

I've been working on a homebrew PS3 port using the PSL1GHT SDK, using RSXGL which implements Opengl 3.1 core. 

### Code Example
Here cgltf.h copys bytes from glb data, which is stored as little-endian, but since the system is big-endian it stores GlbMagic as big-endian while interpreting glb bytes as a big-endian number, causing a mismatch.
```
//cgltf.h
static const uint32_t GlbMagic = 0x46546C67;
...
cgltf_result cgltf_parse(const cgltf_options* options, const void* data, cgltf_size size, cgltf_data** out_data)
{
...
        uint32_t tmp;
        // Magic
        memcpy(&tmp, data, 4);
        if (tmp != GlbMagic) {...}
...
}
```
After fixing cgltf.h with a bunch of __builtin_bswap32(), I could now load glb which behaved the same as gltf files, that is to say still broken. I found that in order for the model to be rendered correctly, I had to swap the bytes of Mesh positions, normals, and indices. I believe the errors comes from the LOAD_ATTRIBUTE_CAST macro function in LoadGLTF.
```
static Model LoadGLTF(const char *fileName)
{
...
//rmodels.c line 5234
    #define LOAD_ATTRIBUTE_CAST(accesor, numComp, srcType, dstPtr, dstType) \
    { \
        int n = 0; \
        srcType *buffer = (srcType *)accesor->buffer_view->buffer->data + accesor->buffer_view->offset/sizeof(srcType) + accesor->offset/sizeof(srcType); \
        for (unsigned int k = 0; k < accesor->count; k++) \
        {\
            for (int l = 0; l < numComp; l++) \
            {\
                dstPtr[numComp*k + l] = (dstType)buffer[n + l];\
            }\
            n += (int)(accesor->stride/sizeof(srcType));\
        }\
    }
...
}
```
I've managed to get glb/gltf files to render correctly by passing  __builtin_bswap32(), perhaps using le32toh() would be better, though my toolchain lack said function.

**Comment by raysan5** [comment]
@nbe1233 raylib was not designed considering Big Endian systems so it could fail in multiple points in that regards... Still, lately some changes by @Peter0x44 where merged to add some BE support.

The reported issue also involves an external library, it should probably reported to https://github.com/jkuhlmann/cgltf

What additional changes are needed to support BE on raylib for gltf loading?

**Comment by Peter0x44** [comment]
I didn't make any changes for that, but parts of the gltf loading code look rather suspicious, and probably assume LE byte order. Everything in a glb is LE.

**Comment by Peter0x44** [comment]
![image](https://github.com/user-attachments/assets/46c601ef-35d7-4ff7-ae83-045c3bfd10ad)
there was a previous discussion about a ps3 port in discord here
I'm not sure if that is simply a BE quirk or the ps3 itself has something to do with it.

Testing powerpc linux with qemu might also help, but I am yet to set that up.

**Comment by Memorix101** [comment]
Currently, we looked into Wii/GameCube on the raylib Discord together, which is BE too. Does the PS3 give you the same results, similar to that, when you load a GLFT/GLB file? 
![image](https://github.com/user-attachments/assets/e597f7ba-de80-451e-84f5-87a9621b118b)
Video: https://github.com/user-attachments/assets/7308d660-5246-499c-b7c4-f8bd0aab5a70

**Comment by nbe1233** [comment]
Yeah I had the same issue. I solved it by replacing LOAD_ATTRIBUTE_CAST() in rmodels.c with this

```

    #define LOAD_ATTRIBUTE_CAST(accesor, numComp, srcType, dstPtr, dstType) \
    { \
        int n = 0; \
        srcType *buffer = (srcType *)accesor->buffer_view->buffer->data + accesor->buffer_view->offset/sizeof(srcType) + accesor->offset/sizeof(srcType); \
        char *dst_type_str = #dstType; \
        bool is_dst_float = false;\
        bool is_dst_ushort = false;\
        if (strcmp(dst_type_str, "float") == 0 ) {is_dst_float = true;} \
        if (strcmp(dst_type_str, "unsigned short") == 0 ) {is_dst_ushort = true;} \
        for (unsigned int k = 0; k < accesor->count; k++) \
        {\
            for (int l = 0; l < numComp; l++) \
            {\
                dstPtr[numComp*k + l] = (dstType)buffer[n + l];\
                if (is_dst_float)\
                {\
                    union { uint32_t u; float f ;} x ;\
                    x.f = dstPtr[numComp*k + l];\
                    x.u = __builtin_bswap32(x.u);\
                    dstPtr[numComp*k + l] = x.f;\
                }\
                if (is_dst_ushort)\
                {\
                    dstPtr[numComp*k + l] = __builtin_bswap16(dstPtr[numComp*k + l]);\
                }\
            }\
            n += (int)(accesor->stride/sizeof(srcType));\
        }\
    }
```

**Comment by nbe1233** [comment]
These are the commits I used to get gltf rendering working
4efa33c
e254c1e

**Comment by raysan5** [comment]
Supporting BE could require many changes to raylib and also external libraries... Have you tried with multiple texture formats or other file-formats?

**Comment by nbe1233** [comment]
With regards to texture formats, I have had issues, but I think it is mostly a driver issue rather than a big-endian issue. Obj 3d models work.

**Comment by raysan5** [comment]
I'm closing this issue, I'm afraid Big Endian is out of scope for raylib at the moment.
